pkg/clients/compute/aks.go

```diff
@@ -10,205 +10,319 @@ You may obtain a copy of the License at
 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
+See the License for the c.Specific language governing permissions and
 limitations under the License.
 */
 
 package compute
 
 import (
 	"context"
-	"strings"
+	"fmt"
+	"time"
 
+	"github.com/Azure/azure-sdk-for-go/services/authorization/mgmt/2015-07-01/authorization"
+	authorizationmgmt "github.com/Azure/azure-sdk-for-go/services/authorization/mgmt/2015-07-01/authorization"
 	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2018-03-31/containerservice"
+	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/date"
 	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
 
-	computev1alpha3 "github.com/crossplane/provider-azure/apis/compute/v1alpha3"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane/provider-azure/apis/compute/v1alpha3"
 	azure "github.com/crossplane/provider-azure/pkg/clients"
-	"github.com/crossplane/provider-azure/pkg/clients/authorization"
 )
 
 const (
 	// AgentPoolProfileName is a format string for the name of the automatically
 	// created cluster agent pool profile
 	AgentPoolProfileName = "agentpool"
 
-	maxClusterNameLen = 31
-)
+	// NetworkContributorRoleID lets the AKS cluster managed networks, but not
+	// access them.
+	NetworkContributorRoleID = "/providers/Microsoft.Authorization/roleDefinitions/4d97b98b-1d4f-4787-a291-c67834d212e7"
```

Comment:
nitpick: does the Azure SDK store this a `const` anywhere?

Reply:
Not sure; this was copied over from the previous implementation. I'd prefer to use a constant, but let's do that when we work on migrating this controller to v1beta1 quality.